@xen-orchestra/fs/src/abstract.js

```diff
@@ -96,6 +101,7 @@ export default class RemoteHandlerAbstract {
     return prefix === '/' ? this : new PrefixWrapper(this, prefix)
   }
 
+  @limit(maximum_operations)
```

Comment:
As I said, the limit should be shared by all methods, example 4 `readFile` + 3 `getSize` + 3 `mkdir`.

Reply:
There's 16 fs method in `@xen-orchestra/fs`
How do you want to share the limit between'em ? 
Unless if we choose to set the limit only for 2 ou 3 methods
- closeFile
- getSize
- list
- mkdir
- mktree
- openFile
- outputFile
- read
- readFile
- rename
- rmdir
- rmtree

Reply:
`limit` is not a decorator: it's a decorator factory, create a single decorator and then apply it to the multiple methods to share the limit:

```js
const sharedLimit = limit(3)

const limitedFoo = sharedLimit(foo)
const limitedBar = sharedLimit(bar)
```